src/functions/postResult/application/save-result-service.ts

```diff
@@ -1,5 +1,48 @@
 import { StandardCarTestCATBSchema } from '@dvsa/mes-test-schema/categories/B';
+import * as mysql from 'mysql2';
+import { IntegrationType } from '../domain/result-integration';
+import { getConnection } from '../../../common/framework/mysql/database';
+import { buildTestResultInsert, buildUploadQueueInsert } from '../framework/database/query-builder';
 
-export const saveTestResult = (testResult: StandardCarTestCATBSchema): Promise<void> => {
-  return Promise.resolve();
+export const saveTestResult = async (testResult: StandardCarTestCATBSchema): Promise<void> => {
+  const connection: mysql.Connection = getConnection();
+
+  const testResultInsert = buildTestResultInsert(testResult);
+  const uploadQueueInsertTars = buildUploadQueueInsert(testResult, IntegrationType.TARS);
+  const uploadQueueInsertRsis = buildUploadQueueInsert(testResult, IntegrationType.RSIS);
+  const uploadQueueInsertNotify = buildUploadQueueInsert(testResult, IntegrationType.NOTIFY);
+
+  return new Promise((resolve, reject) => {
+    connection.beginTransaction((err) => {
+      if (err) {
+        reject(err);
+      }
+      connection.query(testResultInsert, (err) => {
```

Comment:
How comes you're not using your `query` function from `src/common/framework/mysql/database.ts`?

Reply:
Ended up following the docs for transactions on the `mysql` driver docs. I will try using the helper method to clean this up.